hledger/Hledger/Cli/CompoundBalanceCommand.hs

```diff
@@ -61,6 +61,8 @@ compoundBalanceCommandMode CompoundBalanceCommandSpec{..} =
       "show sum of posting amounts (default)"
    ,flagNone ["valuechange"] (setboolopt "valuechange")
       "show change of value of period-end historical balances"
+    ,flagNone ["gain"] (setboolopt "gain")
+      "show gain (present valuation minus cost basis) of period-end historical balances"
```

Comment:
I wonder if we can explain the difference between these more clearly. They are not easy to describe briefly. What would you think of:

  valuechange : show total change of value of period-end historical balance (caused by deposits, withdrawals, market price fluctuations)
  
  gain: show period-end unrealised capital gain/loss (period-end historical balance value minus cost basis)

IIRC I intended both of these to work with period-end dates, as the only really meaningful dates in practice. But did I see that they can be used with other dates (then/now/DATE) ? Am I wrong to say only "period-end" ?

Reply:
`--valuechange` can only be used with period-end, but `--gain` can be used with any valuation. Period end is most likely, but other valuation methods are reasonable too (I want to know the capital gains at a specific point in time, e.g. tax-filing deadline)